template/registry.go

```diff
@@ -81,3 +86,19 @@ func (r *Registry) LineNumber(templateName string, node ast.Node) int {
 	}
 	return 1 + strings.Count(src[:node.Position()], "\n")
 }
+
+// ColNumber computes the column number in the relevant line of input source for the given node
+// within the given template.
+func (r *Registry) ColNumber(templateName string, node ast.Node) int {
+	var src, ok = r.sourceByTemplateName[templateName]
+	if !ok {
+		log.Println("template not found:", templateName)
+		return 0
+	}
+	return 1 + int(node.Position()) - strings.LastIndex(src[:node.Position()], "\n")
+}
+
+// Filename identifies the filename containing the specified template
+func (r *Registry) Filename(templateName string) string {
```

Comment:
Seems a little awkward to return empty string if the template wasn't found. Maybe an ok bool return too?  I see you use that in ColNumber

Reply:
Not sure I follow. Would you like Filename to return an ok bool?

LineNumber and ColNumber will return 0 if the template is not found, which follows the same pattern as returning the empty value.

Or do you mean I should use the ok check within Filename to provide the same logging?

Reply:
Oh, I didn't notice that. I just figured if you used the ability to get an "ok" bool in your usage of it, it might be a good thing to expose, but if this set of functionality returns zero values that seems ok too.

Reply:
I've added the same check to Filename to make the pattern clearer when reading.